### PR TITLE
[Support] Ticket email notifications

### PR DESCRIPTION
**Pull request by GSadee**
| Q             | A
| ------------- | ---
| Bug fix?      | no
| New feature?  | yes
| BC breaks?    | no
| Deprecations? | no
| Fixed tickets | no
| License       | MIT
| Doc PR        | no

Added email notification when creating new Support Ticket

### CODE REVIEW

File: src/Sylius/Component/Support/Adapter/ConfigNotificationRecipientAdapterInterface.php [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Sylius\Component\Support\Adapter;
+
+/**
+ * @author Grzegorz Sadowski <grzegorz.sadowski@lakion.com>
+ */
+interface ConfigNotificationRecipientAdapterInterface
```
Comment (michalmarcinkowski) [inline src/Sylius/Component/Support/Adapter/ConfigNotificationRecipientAdapterInterface.php]: Should be named `NotificationRecipientAdapterInterface`.

File: src/Sylius/Component/Support/Adapter/ConfigNotificationRecipientAdapter.php [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+ 
+namespace Sylius\Component\Support\Adapter;
+
+/**
+ * @author Grzegorz Sadowski <grzegorz.sadowski@lakion.com>
+ */
+class ConfigNotificationRecipientAdapter implements ConfigNotificationRecipientAdapterInterface
```
Comment (michalmarcinkowski) [inline src/Sylius/Component/Support/Adapter/ConfigNotificationRecipientAdapter.php]: Should implement `NotificationRecipientAdapterInterface`.

File: src/Sylius/Component/Support/Provider/NotificationRecipientProvider.php [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+{
+    /**
+     * @var ConfigNotificationRecipientAdapterInterface
+     */
+    private $emailAdapter;
+
+    /**
+     * @param ConfigNotificationRecipientAdapterInterface $emailAdapter
+     */
+    public function __construct(ConfigNotificationRecipientAdapterInterface $emailAdapter)
```
Comment (michalmarcinkowski) [inline src/Sylius/Component/Support/Provider/NotificationRecipientProvider.php]: Should be type hinted to `NotificationRecipientAdapterInterface`.

File: src/Sylius/Component/Support/Provider/NotificationRecipientProvider.php [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+ */
+ 
+namespace Sylius\Component\Support\Provider;
+
+use Sylius\Component\Support\Adapter\ConfigNotificationRecipientAdapterInterface;
+
+/**
+ * @author Grzegorz Sadowski <grzegorz.sadowski@lakion.com>
+ */
+class NotificationRecipientProvider implements NotificationRecipientProviderInterface
```
Comment (michalmarcinkowski) [inline src/Sylius/Component/Support/Provider/NotificationRecipientProvider.php]: Rename class and interface to `NotificationRecipientsProvider`

File: src/Sylius/Component/Support/Adapter/ConfigNotificationRecipientAdapter.php (hunk 2 above)
Comment (pjedrzejewski) [inline src/Sylius/Component/Support/Adapter/ConfigNotificationRecipientAdapter.php]: And should be named: ArrayRecipientsAdapter. I'm thinking that adapter pattern is a bit overkill here, ArrayRecipientsProvider would be enough. WDYT?

**Pull request closed without merging**